### PR TITLE
Fixed extraEnv in Chart Values after upgrade

### PR DESCRIPTION
**Pull request by DevSecOps-Isotrol**
To App 2.37.0 and Chart 1.6.144

**Description**

After upgrade the Chart version we always get an error if extraEnv is used in the Chart Values:

```bash
❯ helm template . --name-template defectdojo --namespace defectdojo --kube-version 1.29 --values override-values.yaml --include-crds --debug
install.go:222: [debug] Original chart version: ""
install.go:239: [debug] CHART PATH: /home/sergio/Documentos/Test/Defectdojo/original/django-DefectDojo/helm/defectdojo

Error: An error occurred while checking for chart dependencies. You may need to run `helm dependency build` to fetch missing dependencies: found in Chart.yaml, but missing in charts/ directory: postgresql, postgresql-ha, redis
helm.go:84: [debug] found in Chart.yaml, but missing in charts/ directory: postgresql, postgresql-ha, redis
helm.sh/helm/v3/pkg/action.CheckDependencies
	helm.sh/helm/v3/pkg/action/install.go:717
main.runInstall
	helm.sh/helm/v3/cmd/helm/install.go:265
main.newTemplateCmd.func2
	helm.sh/helm/v3/cmd/helm/template.go:95
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.8.0/command.go:983
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.8.0/command.go:1115
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.8.0/command.go:1039
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
	runtime/proc.go:271
runtime.goexit
	runtime/asm_amd64.s:1695
An error occurred while checking for chart dependencies. You may need to run `helm dependency build` to fetch missing dependencies
main.runInstall
	helm.sh/helm/v3/cmd/helm/install.go:266
main.newTemplateCmd.func2
	helm.sh/helm/v3/cmd/helm/template.go:95
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.8.0/command.go:983
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.8.0/command.go:1115
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.8.0/command.go:1039
main.main
	helm.sh/helm/v3/cmd/helm/helm.go:83
runtime.main
	runtime/proc.go:271
runtime.goexit
	runtime/asm_amd64.s:1695

❯ echo $?
1
```

**Test results**

Fixed the *_helper.tlp* file it solve the problem:

```bash
 helm dependency build
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "kedacore" chart repository
...Successfully got an update from the "ingress-nginx" chart repository
...Successfully got an update from the "vmware-tanzu" chart repository
...Successfully got an update from the "minio" chart repository
...Successfully got an update from the "kyverno" chart repository
...Successfully got an update from the "mvisonneau" chart repository
...Successfully got an update from the "pact-broker" chart repository
...Successfully got an update from the "trino" chart repository
...Successfully got an update from the "oauth2-proxy" chart repository
...Successfully got an update from the "kiali" chart repository
...Successfully got an update from the "aqua" chart repository
...Successfully got an update from the "defectdojo" chart repository
...Successfully got an update from the "sealed-secrets-web" chart repository
...Successfully got an update from the "open-telemetry" chart repository
...Successfully got an update from the "harbor" chart repository
...Successfully got an update from the "apache-incubator-devlake-helm-chart" chart repository
...Successfully got an update from the "devlake" chart repository
...Successfully got an update from the "community-charts" chart repository
...Successfully got an update from the "istio-official" chart repository
...Successfully got an update from the "argo" chart repository
...Successfully got an update from the "jupyterhub" chart repository
...Successfully got an update from the "istio" chart repository
...Successfully got an update from the "jupyterhub-github" chart repository
...Successfully got an update from the "cilium" chart repository
...Successfully got an update from the "prometheus-community" chart repository
...Successfully got an update from the "grafana" chart repository
...Successfully got an update from the "bitnami" chart repository
...Successfully got an update from the "stable" chart repository
...Successfully got an update from the "vcluster" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 3 charts
Downloading postgresql from repo https://charts.bitnami.com/bitnami
Downloading postgresql-ha from repo https://charts.bitnami.com/bitnami
Downloading redis from repo https://charts.bitnami.com/bitnami
Deleting outdated charts

❯ helm template . --name-template defectdojo --namespace defectdojo --kube-version 1.29 --values override-values.yaml --include-crds --debug
...
❯ echo $?
0
```

The problem is in line 169, nindent of the extraEnv. Changing 8 to 2 it solves the issue.

**Documentation**

...

### CODE REVIEW

**Comment by fcecagno** [comment]
I've tested the patch submitted and can confirm the extra envs are well positioned in the deployment. Without this patch, setting `.Values.extraEnvs` breaks the chart.

File: helm/defectdojo/templates/_helpers.tpl [hunk 1]
```diff
@@ -166,7 +166,7 @@ Create chart name and version as used by the chart label.
           key: postgresql-postgres-password
         {{- end }}
   {{- if .Values.extraEnv }}
-  {{- toYaml .Values.extraEnv | nindent 8 }}
+  {{- toYaml .Values.extraEnv | nindent 2 }}
   {{- end }}
   resources:
     {{- toYaml .Values.django.uwsgi.resources | nindent 10 }}
```
Comment (fcecagno) [inline helm/defectdojo/templates/_helpers.tpl:172]: @DevSecOps-Isotrol would you adjust this indentation as well?
With `nindent 10`:
```
        resources:
                limits:
                  cpu: 2000m
                  memory: 512Mi
                requests:
                  cpu: 100m
                  memory: 256Mi
```
With `nindent 4`:
```
        resources:
          limits:
            cpu: 2000m
            memory: 512Mi
          requests:
            cpu: 100m
            memory: 256Mi
```
I know it doesn't break the chart, but probably this leaded to the error you're fixing in this PR.

**Comment by fcecagno** [comment]
In order to test the patch, create a `values.yaml` with the following content:
```
extraEnv:
  - name: DD_DATABASE_HOST
    valueFrom:
      configMapKeyRef:
        name: my-other-postgres-configmap
        key: cluster_endpoint
```
Then run:
```
helm template -f values.yaml .
```
Without this patch you should see an error, with this patch it should work fine.

File: helm/defectdojo/templates/_helpers.tpl (hunk 1 above)
Comment (DevSecOps-Isotrol) [reply helm/defectdojo/templates/_helpers.tpl:172]: Solved:

```yaml
        resources:
          limits:
            cpu: 2000m
            memory: 512Mi
          requests:
            cpu: 100m
            memory: 256Mi

```

**Review by cneill — COMMENTED** [review]
I can't add a comment on the line itself in GitHub, but the indent on line 145 is likely wrong too. Should be `4`, I believe

```
{{- toYaml .Values.securityContext.djangoSecurityContext | nindent 10 }}
```

**Comment by Maffooch** [comment]
Hmm it looks like the helm chart linter will never pass because `master` was the original base branch. We will need to run the linter manually prior to merge to ensure everything looks good

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Review by blakeaowens — APPROVED** [review]

**Review by cneill — APPROVED** [review]

**Pull request merged**